--- ethtool-3.5/ethtool.c.orig	2014-02-11 15:38:55.000000000 -0500
+++ ethtool-3.5/ethtool.c	2014-02-11 15:39:44.000000000 -0500
@@ -447,6 +447,8 @@ static void dump_supported(struct ethtoo
 		fprintf(stdout, "MII ");
 	if (mask & SUPPORTED_FIBRE)
 		fprintf(stdout, "FIBRE ");
+	if (mask & SUPPORTED_Backplane)
+		fprintf(stdout, "Backplane ");
 	fprintf(stdout, "]\n");
 
 	dump_link_caps("Supported", "Supports", mask, 0);
