--- ethtool-3.5/ethtool.c.bz1003891.1	2014-04-15 13:56:54.000000000 -0400
+++ ethtool-3.5/ethtool.c	2014-04-15 13:57:57.000000000 -0400
@@ -476,6 +476,7 @@ dump_link_caps(const char *prefix, const
 		{ 0, ADVERTISED_2500baseX_Full,     "2500baseX/Full" },
 		{ 0, ADVERTISED_10000baseT_Full,    "10000baseT/Full" },
 		{ 0, ADVERTISED_10000baseKX4_Full,  "10000baseKX4/Full" },
+		{ 0, ADVERTISED_10000baseKR_Full,   "10000baseKR/Full" },
 		{ 0, ADVERTISED_20000baseMLD2_Full, "20000baseMLD2/Full" },
 		{ 0, ADVERTISED_40000baseKR4_Full,  "40000baseKR4/Full" },
 		{ 0, ADVERTISED_40000baseCR4_Full,  "40000baseCR4/Full" },
